crux_http/src/request_builder.rs

```diff
@@ -391,6 +391,22 @@ where
                 .update_app(make_event(self.expectation.decode(resp)));
         });
     }
+
+    /// FIXME: Test and document
+    pub async fn send_async(self) -> crate::Result<Response<ExpectBody>> {
```

Comment:
I'm easy, the `_async` annoys me too. I noticed the `IntoFuture` and wondered how we could use it. I guess the one downside of repurposing that one is that it would be abreaking change for an existing middleware, if any exists.